AI/pyneogame/Agent/PolicyGradient.py

```diff
@@ -101,23 +107,35 @@ def replay_experience(self, batch_size=64, epochs=30):
         states = np.vstack(player_mem[:, 0])
         actions = np.vstack(player_mem[:, 1])
         rewards = player_mem[:, 2]
+        #target = actions 
         target = actions * rewards[:, np.newaxis]
 
         es = EarlyStopping(monitor='val_loss', mode='min',
                            verbose=0, patience=2)
 
+        #self.dnn_model.set_weights(self.save_weights)
         history = self.dnn_model.fit(states,
                                      target,
                                      epochs=epochs,
                                      verbose=0,
                                      batch_size=batch_size,
                                      callbacks=[es],
-                                     validation_split=0.10,
-                                     # sample_weight=np.exp(rewards.astype(float))
+                                     validation_split=0.10
+                                     #sample_weight=np.exp(rewards.astype(float))
                                      )
 
         return history
-
+    
+    def get_action_size(self):
+        return self.actions_size
+    
+    def input_model(self, model):
```

Comment:
Take the model as input to the constructor instead and use the same model creation pattern as done in dqn_test.py